pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -343,6 +346,9 @@ private static void queryRewrite(PinotQuery pinotQuery) {
     // Invoke compilation time functions
     invokeCompileTimeFunctions(pinotQuery);
 
+    // Rewrite Selection list
+    rewriteSelections(pinotQuery.getSelectList());
```

Comment:
We found that the query compilation time for SQL has gone up to 5ms in certain cases (as compared to micro-seconds for PQL). One suspect here is the additional processing like these.

Tagging @siddharthteotia to share the findings.